source/common/config/rds_json.cc

```diff
@@ -0,0 +1,84 @@
+#include "common/config/rds_json.h"
+
+#include "common/common/assert.h"
+#include "common/common/utility.h"
+#include "common/config/json_utility.h"
+#include "common/json/config_schemas.h"
+
+namespace Envoy {
+namespace Config {
+
+void RdsJson::translateWeightedCluster(const Json::Object& json_weighted_clusters,
+                                       envoy::api::v2::WeightedCluster& weighted_cluster) {
+  weighted_cluster.set_runtime_key_prefix(
+      json_weighted_clusters.getString("runtime_key_prefix", ""));
+  const auto clusters = json_weighted_clusters.getObjectArray("clusters");
+  std::transform(clusters.cbegin(), clusters.cend(),
+                 Protobuf::RepeatedPtrFieldBackInserter(weighted_cluster.mutable_clusters()),
+                 [](const Json::ObjectSharedPtr& json_cluster_weight) {
+                   envoy::api::v2::WeightedCluster::ClusterWeight cluster_weight;
+                   JSON_UTIL_SET_STRING(*json_cluster_weight, cluster_weight, name);
+                   JSON_UTIL_SET_INTEGER(*json_cluster_weight, cluster_weight, weight);
+                   return cluster_weight;
+                 });
+}
+
+void RdsJson::translateVirtualCluster(const Json::Object& json_virtual_cluster,
+                                      envoy::api::v2::VirtualCluster& virtual_cluster) {
+  JSON_UTIL_SET_STRING(json_virtual_cluster, virtual_cluster, name);
+  JSON_UTIL_SET_STRING(json_virtual_cluster, virtual_cluster, pattern);
+
+  envoy::api::v2::RequestMethod method{};
```

Comment:
Yep, otherwise we get the compiler complaining that it might be potentially uninitialized (this is an enum, so an int).